examples/nidcpower_source_dc_voltage/teststand_fixture.py

```diff
@@ -39,48 +44,47 @@ def create_nidcpower_sessions(sequence_context: Any) -> None:
         )
         with session_management_client.reserve_sessions(
-            context=pin_map_context,
-            instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DCPOWER,
+            pin_map_context, instrument_type_id=INSTRUMENT_TYPE_NI_DCPOWER
         ) as reservation:
-            for session_info in reservation.session_info:
-                # Leave session open
-                _ = create_session(
-                    session_info,
-                    grpc_device_channel,
-                    initialization_behavior=nidcpower.SessionInitializationBehavior.INITIALIZE_SERVER_SESSION,
-                )
+            # Initialize new sessions on the server, but do not close them.
+            # (Deliberately leak them.)
+            context_manager = reservation.create_nidcpower_sessions(
+                initialization_behavior=SessionInitializationBehavior.INITIALIZE_SERVER_SESSION
+            )
+            _ = context_manager.__enter__()
+
             session_management_client.register_sessions(reservation.session_info)
 
 
 def destroy_nidcpower_sessions() -> None:
     """Destroy and unregister all NI-DCPower sessions."""
     with GrpcChannelPoolHelper() as grpc_channel_pool:
-        session_management_client = nims.session_management.Client(
-            grpc_channel=grpc_channel_pool.session_management_channel
-        )
-        grpc_device_channel = grpc_channel_pool.get_grpc_device_channel(
-            nidcpower.GRPC_SERVICE_INTERFACE_NAME
+        discovery_client = DiscoveryClient(grpc_channel_pool=grpc_channel_pool)
+        session_management_client = SessionManagementClient(
+            discovery_client=discovery_client, grpc_channel_pool=grpc_channel_pool
         )
         with session_management_client.reserve_all_registered_sessions(
-            instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DCPOWER,
+            instrument_type_id=INSTRUMENT_TYPE_NI_DCPOWER
         ) as reservation:
+            # If there are no registered sessions, there is nothing to clean up.
+            if not reservation.session_info:
+                return
+            
             session_management_client.unregister_sessions(reservation.session_info)
 
-            for session_info in reservation.session_info:
-                session = create_session(
-                    session_info,
-                    grpc_device_channel,
-                    initialization_behavior=nidcpower.SessionInitializationBehavior.ATTACH_TO_SERVER_SESSION,
-                )
-                session.close()
+            # Attach to the sessions on the server and explicitly close them.
+            with reservation.create_nidcpower_sessions(
+                initialization_behavior=SessionInitializationBehavior.ATTACH_TO_SERVER_SESSION
+            ) as session_infos:
+                for session_info in session_infos:
+                    session_info.session.close()
```

Comment:
I was a little surprised here that the `with` block didn't automatically close the sessions.

grpc-device returned `new_session_initialized=False`, so nimi-python's `__exit__` method didn't call `close()`. I don't think it checks for `initialization_behavior == AUTO` in `__exit__`.